### DISCUSSION TITLE
Radio Button

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#288] [Component suggestions]
<img width="480" alt="visuele weergave van de radio button component" src="https://github.com/user-attachments/assets/88fdff37-8739-4c53-8826-8d0ffdcc0204">

We willen graag meer leren over de Radio Button component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack]()

---

## Naam
Radio Button

## Doel
Invoerveld voor het kiezen van een enkele optie uit een groep van opties.

## Ook bekend als
Radio

## Gerelateerde componenten
[Radio Group](https://github.com/orgs/nl-design-system/discussions/222), [Checkbox](https://github.com/orgs/nl-design-system/discussions/306), [Switch](https://github.com/orgs/nl-design-system/discussions/293)

## Onderzoek

[Inclusively Hiding & Styling Checkboxes and Radio Buttons - Sara Soueidan](https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/)
Artikel waarin Sara Soueidan beschrijft hoe je custom Checkboxes and Radio Buttons kunt ontwikkelen.

[We've updated the radios and checkboxes - GOV.UK](https://designnotes.blog.gov.uk/2016/11/30/weve-updated-the-radios-and-checkboxes-on-gov-uk/)
Tim Paul legt uit waarom GOV.UK aanpassingen heeft gedaan bij Radio en Checkbox opties.

[Radios - GOV.UK](https://design-system.service.gov.uk/components/radios/)
Onder 'Research on this component' worden bevindingen genoteerd. Er is ook een https://github.com/alphagov/govuk-design-system-backlog/issues/59 waar meer informatie, onderzoek en aanpassingen worden gedeeld.

In een [comment bij Radio Github Discussion](https://github.com/alphagov/govuk-design-system-backlog/issues/59#issuecomment-1358006030) wordt opgemerkt dat een high-contrast gebruiker gebaad is bij een visuele wijziging voor de hover state.

**Comment by jeffreylauwers** [comment]
## Amsterdam

- [Storybook](https://designsystem.amsterdam/?path=/docs/components-forms-radio--docs)
- [Figma](https://www.figma.com/design/9IGm6IdPUYizBNGsUnueBd/Amsterdam-Design-System?node-id=2927-29187)
- [GitHub](https://github.com/Amsterdam/design-system/tree/develop/packages/css/src/components/radio)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 16 27 35](https://github.com/user-attachments/assets/79bc01b2-b4b8-4cc2-b1ea-9d92805a4e91)

Hover
![Scherm­afbeelding 2024-09-30 om 16 27 46](https://github.com/user-attachments/assets/e5806797-1c92-4e02-ac9e-bd5cfe0db29c)

Disabled
![Scherm­afbeelding 2024-09-30 om 16 27 57](https://github.com/user-attachments/assets/063e3f44-cc6d-47d5-be40-95486b77df51)

Focus
![Scherm­afbeelding 2024-09-30 om 16 28 09](https://github.com/user-attachments/assets/d68ebf4c-aaf5-4bf9-ae2e-f8d124c53bd0)

Invalid
![Scherm­afbeelding 2024-09-30 om 16 28 29](https://github.com/user-attachments/assets/a0b79cb5-e492-457c-ac09-db28922b2306)

Checked
![Scherm­afbeelding 2024-09-30 om 16 28 34](https://github.com/user-attachments/assets/645672ee-4ca9-4787-b14a-b10a768653c6)

Checked + Hover
![Scherm­afbeelding 2024-09-30 om 16 28 48](https://github.com/user-attachments/assets/2ba0aedc-a116-4bdc-b58f-9f14aac437d3)

Checked + Disabled
![Scherm­afbeelding 2024-09-30 om 16 28 58](https://github.com/user-attachments/assets/c7768c9a-a244-42ad-971d-87cb152e22b6)

Checked + Invalid
![Scherm­afbeelding 2024-09-30 om 16 29 14](https://github.com/user-attachments/assets/f60342bf-8ed6-4d49-8454-48bf8b5ab816)

---

### Notities

- Documentatie beschikbaar (Storybook)
- Hover state heeft underline en border-width aanpassing.
- Standaard met label.
- Label krijgt underline bij hover state.
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van Amsterdam](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8605510).

**Comment by jeffreylauwers** [comment]
## Den Haag

- [Storybook](https://nl-design-system.github.io/denhaag/?path=/docs/css-radio-button--docs)
- [Figma - Huidig](https://www.figma.com/design/JpoY3waVoQGlLQzQXTL9nn/%F0%9F%8C%BB-Den-Haag-Design-System?node-id=1681-7054&t=jurdUN1yVFEeoE1F-4)
- [Figma - WIP](https://www.figma.com/design/x4RkF6BIdrNZbh7D53NTzB/%F0%9F%92%A0-NLDS---Den-Haag---Bibliotheek?node-id=948-1789)
- [GitHub](https://github.com/nl-design-system/denhaag/tree/main/components/RadioButton)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 16 40 01](https://github.com/user-attachments/assets/647a1dbe-ec25-4083-9c7a-9b0fcf1ee447)

Checked
![Scherm­afbeelding 2024-09-30 om 16 40 27](https://github.com/user-attachments/assets/e5967fb7-2154-47d7-aba6-e66f43310925)

Invalid
![Scherm­afbeelding 2024-09-30 om 16 42 05](https://github.com/user-attachments/assets/6b8b6993-92f1-48c3-8289-836e986eda76)

Disabled
![Scherm­afbeelding 2024-09-30 om 16 42 01](https://github.com/user-attachments/assets/26291fc3-09b9-4fc1-a08d-a3a3325ecf32)

Hover
![Scherm­afbeelding 2024-09-30 om 16 40 07](https://github.com/user-attachments/assets/4d05c9b5-dea5-41e8-8ce7-4b679dd79001)

Focus
![Scherm­afbeelding 2024-09-30 om 16 41 06](https://github.com/user-attachments/assets/dbe91dcd-0ce3-4c3c-b263-7c2e61b1adaf)

Focus visible
![Scherm­afbeelding 2024-09-30 om 16 41 17](https://github.com/user-attachments/assets/3ec19b42-c7de-40a2-9c0a-879f4b1c9c7e)


**Figma - Huidig**

States Form field radio option
![Scherm­afbeelding 2024-02-27 om 14 53 52](https://github.com/nl-design-system/backlog/assets/545207/d877808d-a10f-492f-8624-982017748c4d)

**Figma - WIP**

States Form field radio option
![Scherm­afbeelding 2024-02-28 om 12 12 50](https://github.com/nl-design-system/backlog/assets/545207/24dd1380-07e2-46d3-8023-f51414dd998d)
![Scherm­afbeelding 2024-02-28 om 12 12 56](https://github.com/nl-design-system/backlog/assets/545207/92ae1d4b-9ef2-427e-b33d-02aff5d08605)

---

### Notities

- Storybook en Figma (WIP) zonder label. Figma (Huidig) standaard met label.
- [Zie comment van Rozerin](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8593297) over onderzoek waarbij grotere checkboxes werden ingezet.
- Figma - Huidig component heeft ook een mogelijkheid voor 'Large' variant. Hierbij schaalt het label mee.
- Bekijk ook de [GitHub Discussion - Checkbox group toevoeging van Den Haag](https://github.com/orgs/nl-design-system/discussions/223#discussioncomment-8616643).

**Comment by jeffreylauwers** [comment]
## DSO

[Storybook](https://storybook.dso-toolkit.nl/62.8.4/?path=/story/html-css-selectable--radio)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-09-30 om 16 54 52](https://github.com/user-attachments/assets/f7b781c1-1a94-468c-887f-f4cf7b7f09fb)

Active
![Scherm­afbeelding 2024-09-30 om 16 55 57](https://github.com/user-attachments/assets/b78b4e90-619f-4178-aa06-8808605e354d)

Focus
![Scherm­afbeelding 2024-09-30 om 16 56 12](https://github.com/user-attachments/assets/30b69ca7-0a5c-42df-ae8c-5eae84077565)

Checked
![Scherm­afbeelding 2024-09-30 om 16 55 02](https://github.com/user-attachments/assets/a6ae8667-f660-40bb-82d6-fad72fa549ac)

---

### Notities

- Standaard met label.
- Optie om een 'i-tje' toe te voegen achter het label.
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van DSO](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8605837).

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/checkbox)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-10-01 om 08 50 49](https://github.com/user-attachments/assets/7575b297-bed4-471a-9f60-f57b0aa50398)

Focus
![Scherm­afbeelding 2024-10-01 om 08 48 43](https://github.com/user-attachments/assets/6ccecf8b-c7db-4db6-906a-16148e465c69)

Disabled
![Scherm­afbeelding 2024-10-01 om 08 49 04](https://github.com/user-attachments/assets/bdaf943c-2b88-46ad-a138-ba24157b4fda)

Checked
![Scherm­afbeelding 2024-10-01 om 08 48 43](https://github.com/user-attachments/assets/630e1891-ba95-4bfb-b1d7-cad4e5edbea8)

Uitgebreide radio buttons
Doorgaans wil je korte, simpele opties geven, maar soms vereisen ze wat extra uitleg. Gebruik deze uitgebreide variant als er niet genoeg ruimte is in de hint van de vraag. Let op dat je de korte en uitgebreide varianten niet door elkaar gebruikt.
![Scherm­afbeelding 2024-10-01 om 08 52 02](https://github.com/user-attachments/assets/90bec35c-a8de-454e-a503-c61faaae67fa)

---

### Notities
- Documentatie beschikbaar.
- Standaard met label.
- Kader om Radio button en label, in het geheel 'clickable'.
- Mogelijkheid om extra uitleg te tonen aanvullend op het label.
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van DUO](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8605909).

**Comment by jeffreylauwers** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/story/form-io-components-vanilla-checkbox--default)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-10-01 om 08 56 37](https://github.com/user-attachments/assets/8d0164cb-78ff-4b94-a025-ac1736dcb4c5)

Focus
![Scherm­afbeelding 2024-10-01 om 08 58 33](https://github.com/user-attachments/assets/3caa45fc-091c-4e21-a7ec-80a4ef69a937)

Checked
![Scherm­afbeelding 2024-10-01 om 08 58 11](https://github.com/user-attachments/assets/1683dca4-6f08-4687-8c79-bc15e1304a59)

Extra lang label en description
![Scherm­afbeelding 2024-10-01 om 09 00 50](https://github.com/user-attachments/assets/cc5b730b-52c4-4f38-8485-7f0fad204b61)


---

### Notities

- Standaard met label.
- Kader om checkbox en label, in het geheel 'clickable'.
- Mogelijkheid om extra uitleg te tonen aanvullend op het label.
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van Open Formulieren](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8592426).

**Comment by jeffreylauwers** [comment]
## OpenGemeenten
[Patternlab](https://patternlab.lenteveld.nl/index.html)

---

### Screenshots / Varianten

**Patternlab**

[RadioButton](https://patternlab.lenteveld.nl/patterns/01-molecules-080-Forms-And-Search-RadioButton-RadioButton/01-molecules-080-Forms-And-Search-RadioButton-RadioButton.html)

Default
![Scherm­afbeelding 2024-10-01 om 09 08 51](https://github.com/user-attachments/assets/f325455b-133e-4de1-97e4-f782ffb2b539)

Hover
![Scherm­afbeelding 2024-10-01 om 09 09 02](https://github.com/user-attachments/assets/b2f12602-ae22-4c71-8d82-60e0a74f4f2e)

Focus
![Scherm­afbeelding 2024-10-01 om 09 09 14](https://github.com/user-attachments/assets/640aba96-cf6c-49a8-90f2-7bf009c3e0fa)

Checked
![Scherm­afbeelding 2024-10-01 om 09 09 24](https://github.com/user-attachments/assets/a8194478-66a5-4b2a-bc2d-e2385364ff74)

Checked + Hover
![Scherm­afbeelding 2024-10-01 om 09 09 38](https://github.com/user-attachments/assets/30221ce4-3da9-4f42-8bce-f1a326b515c0)

---

### Notities
- Standaard met label.
- Form field checkbox option is ook voorbij het zichtbare label nog klikbaar.
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van OpenGemeenten](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8607045).

**Comment by jeffreylauwers** [comment]
## Rotterdam

[Figma](https://www.figma.com/file/ZWSC4gCrOXRUR9UX3aoZ8x/Gemeente-Rotterdam-Design-System?type=design&node-id=521-28182&mode=design&t=B6BdewKoEAslWiOU-4)

---

### Screenshots / Varianten

**Figma**

States
![Scherm­afbeelding 2024-02-27 om 16 49 40](https://github.com/nl-design-system/backlog/assets/545207/b07a429d-90d1-45b4-88a9-274676c51ec8)

---

### Notities
- Onder de noemer: Toggles

**Comment by jeffreylauwers** [comment]
## RVO

- [Storybook](https://nl-design-system.github.io/rvo/?path=/docs/componenten-form-radio-button--documentatie)
- [Figma](https://www.figma.com/design/Sj6myBL1Fvot5M1qGxzvEo/ROOS-(RVO-Design-System)?node-id=282-7950)
- [GitHub](https://github.com/nl-design-system/rvo/tree/master/components/form-radio-button)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-10-01 om 11 32 14](https://github.com/user-attachments/assets/8f07f211-c379-4709-b241-be1501685927)


Hover
![Scherm­afbeelding 2024-10-01 om 11 32 20](https://github.com/user-attachments/assets/3b46a0ee-a7fe-4681-96da-2642f859808b)


Active
![Scherm­afbeelding 2024-10-01 om 11 32 29](https://github.com/user-attachments/assets/43d181f4-cfb2-4b98-b648-209b9473b8ec)


Disabled
![Scherm­afbeelding 2024-10-01 om 11 33 07](https://github.com/user-attachments/assets/761381dd-43eb-4594-8660-827da35e24ac)


Focus
![Scherm­afbeelding 2024-10-01 om 11 32 36](https://github.com/user-attachments/assets/a58f78e8-c4b9-46ba-9ee1-a63e02ee797b)


Invalid
![Scherm­afbeelding 2024-10-01 om 11 32 57](https://github.com/user-attachments/assets/92b68c9b-1586-477e-bef0-91d3bee146dd)


Checked
![Scherm­afbeelding 2024-10-01 om 11 32 46](https://github.com/user-attachments/assets/91702324-a0e9-428e-97c8-50a4f7e529fc)


Checked + Hover
![Scherm­afbeelding 2024-10-01 om 11 33 20](https://github.com/user-attachments/assets/726842ef-38e3-4771-9cc4-819a438e610f)


**Figma**
![Scherm­afbeelding 2024-10-01 om 11 34 54](https://github.com/user-attachments/assets/e65c8b15-72a8-41e2-896c-352c0e067869)

---

### Notities

- Standaard met label.
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van RVO](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8607243).

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-radio-button--docs)
- [Figma](https://www.figma.com/design/UXIHcIurAD8hyoBWx4hDBV/NLDS---Gemeente-Utrecht---Bibliotheek?node-id=948-1789)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/radio-button)

---

### Screenshots / Varianten

**Storybook**

Veel verschillende states beschikbaar in Storybook. Maar niet elke state zorgt voor een wisseling in vormgeving. Vandaar dat er maar enkele states meegenomen in screenshots.

Default
![Scherm­afbeelding 2024-10-01 om 12 18 16](https://github.com/user-attachments/assets/081a35ef-35df-4056-ac93-6ee52abd1a62)

Hover
![Scherm­afbeelding 2024-10-01 om 12 18 21](https://github.com/user-attachments/assets/1e68043a-9fa9-4d61-ac17-3594148ceca8)

Active
![Scherm­afbeelding 2024-10-01 om 12 18 40](https://github.com/user-attachments/assets/906c6262-285f-4c99-b4e6-2fda87684071)

Focus
![Scherm­afbeelding 2024-10-01 om 12 18 26](https://github.com/user-attachments/assets/06258f6f-3759-4037-bb1b-4c8acd16c6a9)

Focus Visible
![Scherm­afbeelding 2024-10-01 om 12 18 30](https://github.com/user-attachments/assets/bdb5421c-2221-4394-8450-1802adb5fe0d)

Disabled
![Scherm­afbeelding 2024-10-01 om 12 18 45](https://github.com/user-attachments/assets/d66702e2-0e28-402d-aa60-3af4a4055603)

Invalid
![Scherm­afbeelding 2024-10-01 om 12 19 22](https://github.com/user-attachments/assets/561e606c-ff6c-41cd-aeba-0d73fee0972a)

Checked
![Scherm­afbeelding 2024-10-01 om 12 18 51](https://github.com/user-attachments/assets/b7ed7197-72c7-46a8-ab0d-dc0028659910)

Checked + Hover
![Scherm­afbeelding 2024-10-01 om 12 18 59](https://github.com/user-attachments/assets/bda5f7c8-5acb-49ab-8084-8b009cd233fe)

Checked + Active
![Scherm­afbeelding 2024-10-01 om 12 19 16](https://github.com/user-attachments/assets/570c7ca2-6798-4072-85fc-3bfc16bb4a72)

Checked + Disabled
![Scherm­afbeelding 2024-10-01 om 12 19 13](https://github.com/user-attachments/assets/2dc79e0d-4b5e-4edb-8d28-26e5a92a18e0)

Checked + Invalid
![Scherm­afbeelding 2024-10-01 om 12 19 27](https://github.com/user-attachments/assets/5b26d526-6bdb-4b6f-bb86-e5142a4f8f24)


**Figma**
States van Radio button als onderdeel van Form Field Radio Option

![Scherm­afbeelding 2024-02-28 om 12 13 14](https://github.com/nl-design-system/backlog/assets/545207/6836541a-e38d-4d1a-ba88-a9e2e61d9d6b)
![Scherm­afbeelding 2024-02-28 om 12 13 19](https://github.com/nl-design-system/backlog/assets/545207/29f933d4-f675-45f3-bd4d-e67149f3f441)



---

### Notities

- Documentatie beschikbaar (Storybook)
- Zonder label. 
- Bekijk ook de [GitHub Discussion - Radio group toevoeging van Utrecht](https://github.com/orgs/nl-design-system/discussions/222#discussioncomment-8607551).

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Radio Button op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=radio+button)

**Comment by petergoes** [comment]
## ✨ Dit component is nu Community ✨


Help je mee hem door de Candidate stappen te krijgen?
[Radio Button op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=radio+button)
